src/formatDate.js

```diff
@@ -50,7 +50,29 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDate = date.split(`${fromFormat[fromFormat.length - 1]}`);
+  const resultDate = [];
+  const objDate = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    objDate[fromFormat[i]] = newDate[i];
+  }
+
+  if (objDate.hasOwnProperty('YY')) {
+    objDate.YYYY = objDate.YY < 30
+      ? '20' + objDate.YY
+      : '19' + objDate.YY;
+  }
+
+  if (objDate.hasOwnProperty('YYYY')) {
+    objDate.YY = objDate.YYYY.slice(2);
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    resultDate.push(objDate[toFormat[i]]);
+  }
+
+  return resultDate.join(`${toFormat[toFormat.length - 1]}`);
```

Comment:
new separator is already a string, so you dont need to use formated str `${x}` )
```suggestion
  return resultDate.join(toFormat[toFormat.length - 1]);
```